dds_web/api/project.py

```diff
@@ -473,3 +479,43 @@ def get(self):
             research_users.append(user_info)
 
         return {"research_users": research_users}
+
+
+class ProjectAccess(flask_restful.Resource):
+    """Renew project access for users."""
+
+    @auth.login_required(role=["Unit Admin", "Unit Personnel", "Project Owner"])
+    @logging_bind_request
+    @dbsession
+    @user_required
+    @project_optional
+    @renew_access_required
```

Comment:
I must say I'm not a fan of using decorators this way. I think it makes following the code logic very difficult. But that's entirely up to you.

Reply:
What are you suggesting that I change? We need the functionality for multiple endpoints, but we could change them to function calls if you prefer that